PokerGame/PokerEngine/Poker.cs

```diff
@@ -28,17 +34,79 @@ public static List<Card> CalculateWinningHand(List<Card> handOne, List<Card> han
             return CalculateHandWithHighestCardAndKicker(handOne, handTwo);
         }
 
+        private static List<Card> CalculateHandHasWithTheBestStraight(List<Card> handOne, List<Card> handTwo)
+        {
+            Rank rankOfHandOneHighCard;
+            var handOneHasStraight = hasStraight(handOne, out rankOfHandOneHighCard);
+
+            Rank rankOfHandTwoHighCard;
+            var handTwoHasStraight = hasStraight(handTwo, out rankOfHandTwoHighCard);
+
+            if (handOneHasStraight && handTwoHasStraight)
+            {
+                return handWinsWhenEqualRankButHigherCard(handOne, handTwo, rankOfHandOneHighCard, rankOfHandTwoHighCard);
+            }
+
+            if (handOneHasStraight)
+            {
+                return handOne;
+            }
+            if (handTwoHasStraight)
+            {
+                return handTwo;
+            }
+            return null;
+        }
+
+        private static bool hasStraight(List<Card> hand, out Rank rankOfHighestCardInStraight)
+        {
+            var handOrderedByHighestCard = hand
```

Comment:
This is ambiguous as to how it's ordered. "OrdereByHighestCard" suggests descending order, but it's actually ascending order.  Make it more clear please.